### ISSUE TITLE
Add the ability to hide IdPs from linked-accounts endpoint

### ISSUE THREAD

**Issue by btarhan** [keycloak/keycloak#24117]
### Description

Currently, the linked-accounts endpoint returns all IdPs on a realm (assuming there is not a GUI Order collision). We would like to make this endpoint more precise, to only return the IdPs that a given user is able to link to, or already linked to. We do this pruning ourselves via frontend customizations today, but would prefer the linked-accounts endpoint be more precise with the data it returns.

**Comment by pedroigor** [comment]
@ssilvert @mposolda It should be a simple change if we add a parameter to control whether unlinked brokers are returned in the response.

Other than that, we can solve this by customizations using the Account REST API or allow configuring on a per-broker basis if users are allowed to link accounts through the account console.

**Comment by pedroigor** [comment]
Today we have two settings to control how brokers are shown in user-facing flows:

* `Hide on login page`, this setting will not show a specific broker in login pages
* `Account link only`, this setting will prevent users from authenticating using a specific broker and also do not show it in login pages

However, I think the two settings above are a bit confusing because the `Account link only` setting is also hiding the broker from the login page (same as `Hide on login page`) but the main behavior introduced by this setting is to prevent users from authenticating using the broker. For instance, suppose you have set:

* `Account link only=ON`
* `Hide on login page=OFF`

By reading this configuration, I would expect the login page to show the broker even though the user will not be able to authenticate using this broker. However, the current behavior on the login page is that the broker is not shown even though the configuration says otherwise.

I think all the above is a separate issue but related to this one given that those settings will be even more confusing if we decide to introduce the behavior herein being requested.

For this specific issue, I think we are missing a setting to filter the brokers that should be available for linking in the account console. We could have a `Allow account linking ` setting to decide whether or not brokers are available for linking accounts therefore filtering them out when at the `Linked accounts` section in the account console.

That will probably mean changing the label of `Account linking` to `Allow authenticating` (or something better) to make more explicit that this setting is mainly about enforcing that users cannot authenticate through the broker.

**Comment by sguilhen** [comment]
I don't think we should ever display a broker that can't be used for authentication - this part to me works fine. Hide on login has a different intent, although to the final user the experience might look like it is the same. AFAIK you can use things like `login_hint` to direct users to an IDP that has `HIDE_ON_LOGIN=true`, but the same mechanism won't make an IDP with `ACCOUNT_LINK_ONLY=true` available for authentication.
 
Not sure how adding a new flag would make things easier for us - if we set `Allow auth` to false, then we presume it is an IDP that can be used for account linking only. So this would automatically set the new `Allow account linking` property to `true` in this case, right? This setting should be editable only if the IDP is available for authentication, in which case it can also work for linking accounts.

**Comment by pedroigor** [comment]
My point is that turning settings ON/OFF based on others is not a good experience, for this particular case. As I mentioned before, and as you reiterated, it does not make sense to have `Allow auth` to `false` and `Hide on login page` to `false` but that is exactly what you read from the configuration. The administrator can then fix/update the configuration accordingly.

In addition to that, controlling this relationship between settings introduces additional/unnecessary complexity. I mean, considering what I think is a better UX.

Note that the `Allow account linking` has a different meaning than the current `Account link only`.

> Not sure how adding a new flag would make things easier for us - if we set Allow auth to false, then we presume it is an IDP that can be used for account linking only. So this would automatically set the new Allow account linking property to true in this case, right? 

But how would you disable the broker to not show in the account console? That is the reason behind the new `Allow account linking` setting I'm suggesting ....

**Comment by sguilhen** [comment]
But what is the purpose of an IDP that is not available for login and also is not available for linking?

**Comment by mposolda** [comment]
Few options and pointers here:

- +1 that having `Account link only=ON` and `Hide on login page=OFF` is confusing. Could we maybe hide the option `Hide on login page` in case that `Account link only` is ON? If the broker is for `Account link only`, then it is implicitly hidden on the login page, so there is no need to hide the option? However I am not sure if this can be done as IDP pages in admin console are dynamically rendered?
 
- Maybe we can have validation added to the IDP? Something like `ComponentFactory.validateConfiguration` method? AFAIK there is nothing like that available yet for IDP (at least not seeing anything on `IdentityProviderFactory` ). In that case, it won't be allowed to have `Hide on login page` set to OFF in case that `Account link only` is ON.
 
- With validation, we will be also able to fail validation if `Account link only` is ON and `Allow account linking` is OFF (as it means that broker cannot be used for authentication and cannot be used even for account-linking, so likely not much point in having such broker as mentioned by @sguilhen ? 

- Not sure about rename existing flag to `Account link only` to `Allow authentication` as it will have exactly opposite meaning, which has some risk for migration / backwards compatibility bugs...
